docs/account/user-session-management.md

```diff
@@ -0,0 +1,13 @@
+# User session management
+
+To reduce the risk of unauthorized access to your account, Codacy sets automatic expiration timeouts for every session. When a session expires, Codacy invalidates the session both on the client and server side, and creates a new session.
+
+If your Git provider session is still valid by the time Codacy creates a new session, you will be automatically logged in again. In some specific scenarios, you might need to select your Git provider before logging in again.
```

Comment:
have you seen the design for the timeout warning? https://www.figma.com/design/qxBcHZXqniAZKt8eXZj7sv/Design-Template-%5BDuplicate-this%5D-(Copy)?node-id=1-171&t=WckeYPBtjFdpjJgy-1
We still haven't implemented it, but ideally we would either notify them before log them out, or show them the action to log in back

Reply:
@andrzej-janczak can explain better probably 😄

Reply:
> have you seen the design for the timeout warning? https://www.figma.com/design/qxBcHZXqniAZKt8eXZj7sv/Design-Template-%5BDuplicate-this%5D-(Copy)?node-id=1-171&t=WckeYPBtjFdpjJgy-1

@inesgomas Thank you for flagging it! 🙏
I just checked the design. 👍 
Anyway, I don't think it's relevant to include in the documentation all the details about the timeout warning that pops up to the user before the session expires, as it's a very common behavior within the session expiration flow. WDYT?

@andrzej-janczak Thoughts?

Reply:
Sounds about right. I was just concerned about the "auto log in" because I think we will show the popup instead and let them choose if they want to log in or not. I got that idea from the designs. But maybe we don't need to worry about that detail.

Reply:
After today's daily: it seems we won't implement the popup in this version, so skip this comment ☝️

Reply:
Yep! 👌

Reply:
@inesgomas @andrzej-janczak 

So, getting back to this conversation again, now that we will implement the popup... 😉 

If we won't have auto-login, I would just remove this sentence. I don't think we need to include the popup flow in the documentation. WDYT?

```suggestion
```

Reply:
Sounds good!